app/src/main/java/com/example/trashminder/utils/AlarmReceiver.kt

```diff
@@ -79,13 +89,85 @@ class AlarmReceiver : BroadcastReceiver() {
             }
         }
 
-        notifications.setRepetitiveAlarm(
-            cal.timeInMillis,
-            context,
-            cal.timeInMillis.toInt(),
-            type,
-            repetition
+        val formatter = SimpleDateFormat("dd/MM/yyyy HH:mm", Locale.FRANCE)
+        val dateString = formatter.format(Date(cal.timeInMillis))
+
+        if (idReminder != null) {
+            updateReminder(date, repetition, type, dateString, idReminder)
+
+            notifications.setRepetitiveAlarm(
+                cal.timeInMillis,
+                context,
+                cal.timeInMillis.toInt(),
+                type,
+                repetition,
+                dateString,
+                idReminder + 1
+            )
+        }
+
+        Log.d("Update", "dateString = $dateString")
+    }
+
+    private fun updateReminder(
+        date: String?,
+        repetition: String?,
+        type: String?,
+        dateString: String,
+        idReminder: Int
+    ) {
+        val trash = mutableMapOf(
+            "date" to date,
+            "id" to idReminder,
+            "repetition" to repetition,
+            "type" to type
+        )
+
+        Log.d("Update", "dateString = $dateString")
+
+        val newTrash = mutableMapOf(
+            "date" to dateString,
+            "id" to idReminder + 1,
+            "repetition" to repetition,
+            "type" to type
         )
+
+        firestore.collection("users").whereArrayContains("reminders", trash).get()
```

Comment:
This chunk of code needs more attention. Quite hard to read. Anything you can do to improve it? Maybe split it in smaller chunks? Also, if Logs are there only for your testing and they do not provide critical additional data, they can be removed